From d341679994d9dc525da3a4b94f3f88032f562c8c Mon Sep 17 00:00:00 2001
From: Thomas Langer <thomas.langer@intel.com>
Date: Tue, 22 Oct 2019 18:37:52 +0200
Subject: [PATCH] PONRTSYS-5112: Ignore bootloader arguments for TEP kernel

The generic functions to access the fw_args are used in relocate.c
and will trigger crashes, as the values are not provided correctly.
Just ignore them and set them to 0 to avoid the problem is okay,
as the code in prom.c is anyway overwriting the cmdline.
---
 arch/mips/kernel/head.S | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/mips/kernel/head.S b/arch/mips/kernel/head.S
index b075fb2463b3..b863e4d12c62 100755
--- a/arch/mips/kernel/head.S
+++ b/arch/mips/kernel/head.S
@@ -93,6 +93,14 @@ NESTED(kernel_entry, 16, sp)			# kernel entry point
 	jr	t0
 0:
 
+#ifdef CONFIG_SOC_TYPE_GRX500_TEP
+	/* ignore arguments from bootloader for TEP */
+	move		a0, zero
+	move		a1, zero
+	move		a2, zero
+	move		a3, zero
+#endif
+
 #ifdef CONFIG_USE_OF
 #ifdef CONFIG_MIPS_RAW_APPENDED_DTB
 	PTR_LA		t2, __appended_dtb
